src/server/game_server.rs

```diff
@@ -126,24 +144,49 @@ impl GameServer {
         }
     }
     
-    pub fn on_new_attack(&mut self, attack: EntityAttack) {
+    pub fn on_new_attack(&mut self, attack: EntityAttack, player_id: usize) {
```

Comment:
Why do we need `player_id ` ? I think we need to remove it.

It's fine if the player who killed a monster receives the killing action. It's the server who is in charge of knowing what to kill, not the client.